core/src/test/java/io/aiven/klaw/TopicRequestValidatorImplIT.java

```diff
@@ -0,0 +1,276 @@
+  }
+
+  @Test
+  @Order(3)
+  public void isValidTestTopicName() {
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(1001);
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(false);
+    when(topicControllerService.getUserName()).thenReturn(KWUSER);
+    when(topicControllerService.getEnvsFromUserId(any())).thenReturn(List.of("1"));
+
+    addTopicRequest.setTopicname("");
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(2);
+    assertThat(violations.toString()).contains("Failure. Please fill in a valid topic name.");
+    assertThat(violations.toString()).contains("Invalid topic name");
+
+    addTopicRequest.setTopicname("testtopic$@$@#$"); // with special characters
+    when(topicControllerService.getTopicFromName(anyString(), anyInt()))
+        .thenReturn(Collections.emptyList());
+    when(topicControllerService.getEnvDetails(anyString()))
+        .thenReturn(utilMethods.getEnvListsIncorrect1().get(0));
+    violations = validator.validate(addTopicRequest);
+    assertThat(violations.toString()).contains("Invalid topic name");
+  }
+
+  @Test
+  @Order(4)
+  public void isValidTestVerifyTenantConfigExists() {
+    Integer tenantId = 1;
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(1001);
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(false);
+    when(topicControllerService.getUserName()).thenReturn(KWUSER);
+    when(topicControllerService.getEnvsFromUserId(any())).thenReturn(List.of("1"));
+    when(commonUtilsService.getTenantId(any())).thenReturn(tenantId);
+    when(topicControllerService.getSyncCluster(anyInt()))
+        .thenThrow(new RuntimeException("Sync cluster not configured"));
+
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
+    assertThat(violations.toString())
+        .contains("Failure. Tenant configuration in Server config is missing. Please configure.");
+  }
+
+  @Test
+  @Order(5)
+  public void isValidTestTopicOwnedByDifferentTeam() {
+    Integer tenantId = 101;
+    Topic topic = utilMethods.getTopic("testtopic");
+
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(1001);
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(false);
+    when(topicControllerService.getUserName()).thenReturn("superadmin");
+    when(topicControllerService.getEnvsFromUserId(any())).thenReturn(List.of("1"));
+    when(commonUtilsService.getTenantId(any())).thenReturn(tenantId);
+    when(topicControllerService.getTopicFromName(anyString(), anyInt())).thenReturn(List.of(topic));
+
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
```

Comment:
```suggestion
    assertThat(violations).hasSize(1);
```
assertJ has `hasSize` it's better to use it here

Reply:
Agree, updated.